src/api/v1/users/register/confirm.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Registration endpoints for visitors
+ */
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+import * as Joi from 'joi';
+import { id, response } from '../schema';
+import {
+  Visitors,
+  CommunityBusinesses,
+  CbAdmins,
+  Users,
+  Volunteers,
+} from '../../../../models';
+import * as QRCode from '../../../../services/qrcode';
+import * as PdfService from '../../../../services/pdf';
+import { EmailTemplate } from '../../../../services/email/templates';
+import { RegisterConfirm } from '../../types';
+import { RoleEnum, Roles } from '../../../../auth';
+
+export default [
+  {
+    method: 'POST',
+    path: '/users/register/confirm',
+    options: {
+      description: 'Confirm adding a new role to an existing user',
+      auth: false,
+      validate: {
+        payload: {
+          organisationId: id.required(),
+          userId: id.required(),
+          role: Joi.alternatives(RoleEnum.VOLUNTEER, RoleEnum.VISITOR).required(),
+          token: Joi.string().length(64).required(),
+        },
+      },
+      response: { schema: response },
+    },
+    handler: async (request: RegisterConfirm, h: Hapi.ResponseToolkit) => {
+      const {
+        payload: { role, userId, token, organisationId },
+        server: { app: { EmailService, knex } },
+      } = request;
+      const cb = await CommunityBusinesses.getOne(knex, { where: { id: organisationId } });
+      const user = await Users.getOne(knex, { where: { id: userId } });
+
+      const [admin] = await CbAdmins.fromOrganisation(knex, { id: organisationId });
+
+      /* istanbul ignore next */
+      if (!admin) {
+        // Because of a change in testing method, this is now unreachable
+        // (and was always functionally impossible), but is kept **just
+        // in case**
+        throw Boom.badData('No associated admin for this organisation');
+      }
+
+      // Checks
+      const currentRoles = await Roles.fromUser(knex, user);
+      // check role doesnt exist on user account
+      if (currentRoles.some((x) => x.role === role)) {
```

Comment:
Roles.userHas does something slightly different:

```js
{
	role: RoleEnum
	userId: number,
    organisationId: number
} => boolean
```

The above grabs a list of all the roles a user has so that we can check if they are for the same organization.